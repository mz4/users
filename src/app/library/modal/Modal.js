import { useRef } from 'react';
import { useClickOutside } from '../../hooks/index';
import { Button } from '../index';
import styles from './Modal.module.scss';

export const Modal = ({ action, children, modalType }) => {
  const type = {
    primary: styles.primary,
    secondary: styles.secondary
  };
  const modalRef = useRef();
  useClickOutside(modalRef, action);
  return (
    <div className={type[modalType]}>
      <div ref={modalRef} className={styles.modalBody}>
        <div className={styles.header}>
          <Button type="secondary" text="Close" actionButton={action} />
        </div>
        {children}
      </div>
    </div>
  );
};
