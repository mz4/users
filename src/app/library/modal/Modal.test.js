import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { Modal } from './Modal';

describe('Render modal component', () => {
  const handleClick = jest.fn();
  const children = <div className={'modalBodyChildren'}>modal body</div>;
  const modal = (
    <Modal
      action={handleClick}
      children={<div className={children}>modal body</div>}
    />
  );
  it('Create a snapshot', () => {
    const component = renderer.create(modal);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Render modal component and children', () => {
    const { container } = render(modal);
    const elementModalContainer =
      container.getElementsByClassName('modalContainer');
    const elementModalBody = container.getElementsByClassName('modalBody');
    const elementModalBodyChildren =
      container.getElementsByClassName('modalBodyChildren');
    const elementHeader = container.getElementsByClassName('header');
    expect(elementModalContainer).toBeDefined();
    expect(elementModalBody).toBeDefined();
    expect(elementModalBodyChildren).toBeDefined();
    expect(elementHeader).toBeDefined();
  });
  it('Call a function on button click', () => {
    render(modal);
    fireEvent.click(screen.getByText('Close'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
