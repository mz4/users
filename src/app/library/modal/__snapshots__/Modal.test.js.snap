// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render modal component Create a snapshot 1`] = `
<div>
  <div
    className="modalBody"
  >
    <div
      className="header"
    >
      <button
        className="secondary"
        onClick={[MockFunction]}
      >
        Close
      </button>
    </div>
    <div
      className={
        <div
          className="modalBodyChildren"
        >
          modal body
        </div>
      }
    >
      modal body
    </div>
  </div>
</div>
`;
